src/main/java/bio/terra/service/snapshotbuilder/utils/HierarchyQueryBuilder.java

```diff
@@ -38,22 +38,31 @@ public Query generateQuery(SnapshotBuilderDomainOption domainOption, int concept
     var conceptRelationship =
         TableVariable.forPrimary(TablePointer.fromTableName(ConceptRelationship.TABLE_NAME));
     var relationshipId = conceptRelationship.makeFieldVariable(ConceptRelationship.RELATIONSHIP_ID);
-    var conceptId1 = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_1);
-    var conceptId2 = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_2);
+    var parentId = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_1);
+    var childId = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_2);
     var child =
         TableVariable.forJoined(
-            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, conceptId2);
+            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, childId);
     var parent =
         TableVariable.forJoined(
-            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, conceptId1);
+            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, parentId);
     FieldVariable conceptName = child.makeFieldVariable(Concept.CONCEPT_NAME);
     FieldVariable conceptCode = child.makeFieldVariable(Concept.CONCEPT_CODE);
 
+    // To get the total occurrence count for a child concept, we need to join the child through the
+    // ancestor table to find all of its children. We don't need to use a left join here
+    // because every concept has itself as an ancestor, so there will be at least one match.
+    var conceptAncestor =
```

Comment:
There's some overlap here with `SearchConceptsQueryBuilder` as well, the only difference is`childId`. Possible place where things can be conjoined